go/vt/vtctl/reparentutil/replication.go

```diff
@@ -181,8 +182,9 @@ func StopReplicationAndBuildStatusMaps(
 		logger.Infof("getting replication position from %v", alias)
 
 		_, stopReplicationStatus, err := tmc.StopReplicationAndGetStatus(groupCtx, tabletInfo.Tablet, replicationdatapb.StopReplicationMode_IOTHREADONLY)
-		switch err {
-		case mysql.ErrNotReplica:
+
+		switch {
+		case vterrors.Code(err) == vtrpc.Code_INTERNAL && strings.Contains(err.Error(), vterrors.NoReplicationStatusStr):
```

Comment:
@harshit-gangal @vmg this is the line Manan is referring to in his question.
We seem to be already calling `FromGRPC` and `ToGRPC` in several places so that is not a concern.